packages/pluggableWidgets/document-viewer-web/src/utils/useFileURL.ts

```diff
@@ -0,0 +1,60 @@
+import { DocumentViewerContainerProps } from "../../typings/DocumentViewerProps";
+import { useMemo } from "react";
+import { ValueStatus } from "mendix";
+
+type UseFileUrlHookResult =
```

Comment:
I don't understand why it's not combined in one type? like, 
`{
          loading: boolean;
          data: null | string;
          error: null | string;
}`
If there's any reason to keep it separated just let me know.

Reply:
Yeah, the reason for doing this is TypeScript, this is called [Discriminated unions](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) int ts. Actually I took this pattern from Apollo. Main idea is to have better types and behavior depending on presence of error.

Reply:
I agree with you, but that's when you have different property keys for different type of items. in your case it's the same shape. In the documentation it shows how it's separating `Circle` and `Square` because one has `sideLength` the other one has `radius`. But again, you have 3 different types with same keys. `{loading, data, error}`. Narrowing it's not about `value type`, but about `key`.

Reply:
Actually it is, here short example
```
typeof result.data // => data: null | string

if (result.error !== null) {
  return result.error;
}

typeof result.data // => data: string
```

Reply:
@raffiar, added readable name for types